mongo_connector/doc_managers/elastic_doc_manager.py

```diff
@@ -118,11 +137,20 @@ def update(self, document_id, update_spec, namespace, timestamp):
         """
         self.commit()
         index, doc_type = self._index_and_mapping(namespace)
-        document = self.elastic.get(index=index, doc_type=doc_type,
-                                    id=u(document_id))
-        updated = self.apply_update(document['_source'], update_spec)
-        # _id is immutable in MongoDB, so won't have changed in update
-        updated['_id'] = document['_id']
+
+        document = None
+        if doc_type in self.routing:
+            # we need to manually retrieve the parent id from mongo
+            db, collection = self.command_helper.revert_namespace(namespace)
+            updated = self.mongo[db][collection].find_one(document_id)
+            if updated is None:
+                return # if the parent was deleted we can't do anything about it
```

Comment:
I'm not sure I understand this part; this method is called whenever a document is updated. Why are we retrieving the most recent version of the document from MongoDB rather than applying the update as usual? What's the significance of `doc_type in self.routing` being `True`?